minidump/src/minidump.rs

```diff
@@ -3917,6 +3923,26 @@ impl<'a> MinidumpStream<'a> for MinidumpLinuxProcLimits<'a> {
     }
 }
 
+impl<'a> MinidumpStream<'a> for MinidumpSoftErrors<'a> {
+    const STREAM_TYPE: u32 = MINIDUMP_STREAM_TYPE::MozSoftErrors as u32;
+
+    fn read(
+        bytes: &'a [u8],
+        _all: &'a [u8],
+        _endian: scroll::Endian,
+        _system_info: Option<&MinidumpSystemInfo>,
+    ) -> Result<MinidumpSoftErrors<'a>, Error> {
+        let json_str = std::str::from_utf8(bytes).map_err(|_| Error::DataError)?;
```

Comment:
Note: I would have rather parsed this all the way through to a `serde_json::Value` (since it is an error for this to be anything but valid JSON), but `serde_json` is not already a dependency of this crate, and I didn't think it worth adding it for this.

Reply:
I think this is a good choice given this is the base crate and has traditionally only contained the raw unparsed contents of a minidump. We put semantics on them in minidump-processor.

Reply:
Fair enough :) JSON is weird sometimes so it's hard to know where to draw the line -- It kind-of straddles the boundary between "having semantics" and "just being a wire format".